File: java-backend/mysql/service/OrderService.java
package com.example.universaldb.service;

import com.example.universaldb.model.OrderDTO;
import com.example.universaldb.repository.MySqlOrderRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class OrderService {

    private final MySqlOrderRepository repository;

    public OrderService(MySqlOrderRepository repository) {
        this.repository = repository;
    }

    public List<OrderDTO> getOrdersForUser(int userId) {
        return repository.findOrdersByUser(userId);
    }
}

File: php-backend/mysql/OrderService.php
<?php

require_once 'OrderRepository.php';

class OrderService
{
    private OrderRepository $repository;

    public function __construct(OrderRepository $repository)
    {
        $this->repository = $repository;
    }

    public function getOrdersForUser(int $userId): array
    {
        return $this->repository->findOrdersByUser($userId);
    }
}
